biom/table.py

```diff
@@ -1795,39 +1797,23 @@ def sort_order(self, order, axis='sample'):
         O2	1.0	0.0	4.0
 
         """
-        md = []
-        vals = []
+        fancy = np.array([self.index(i, axis=axis) for i in order], dtype=int)
```

Comment:
These are index positions so content of the matrix doesn't matter. Forcing `int` as numpy by default does `float`.